pkg/apply/layerApplier.go

```diff
@@ -606,6 +649,31 @@ func (a KubectlLayerApplier) PruneIsRequired(ctx context.Context, layer layers.L
 	return pruneRequired, pruneHrs, nil
 }
 
+// processUpdateVersionAnnotation checks for kraan.updateVersion annotation set to true and updates values to include current version.
+func (a KubectlLayerApplier) processUpdateVersionAnnotation(layer layers.Layer, source *helmctlv2.HelmRelease) error {
+	annotations := source.GetAnnotations()
+	for k, v := range annotations {
+		if k == "kraan.updateVersion" && v == "true" {
+			a.logTrace("values for chart with annotation", layer, append(logging.GetObjKindNamespaceName(source), "values", logging.LogJSON(source.Spec.Values))...)
+			var values map[string]interface{}
+			if source.Spec.Values != nil {
+				err := json.Unmarshal(source.Spec.Values.Raw, &values)
+				if err != nil {
+					return errors.WithMessagef(err, "%s - failed convert values to json string", logging.CallerStr(logging.Me))
+				}
+			}
+			values["kraanVersion"] = layer.GetSpec().Version
```

Comment:
This would effectively set the values[kraanVersion] to the layer's version? But then now I have to figure out how to generate a new version for every build. Can we not use a Hash or Timestamp?

Reply:
If we use a generated value it will cause the annotated HR to be redeployed everytime reconcile runs, we want it to run when we deem it is needed and updating the version of the addonslayer was what I thought we discussed?

Reply:
see https://github.com/fidelity/kraan/blob/acca23d151bcf781546776013dda5bb6908e104f/docs/user-guide.md#versions

Reply:
thank you! LGTM